core/src/main/java/com/bumble/appyx/core/node/ParentNode.kt

```diff
@@ -212,6 +221,48 @@ abstract class ParentNode<Routing : Any>(
         }
     }
 
+    protected suspend inline fun <reified T> attachWorkflow(
+        crossinline action: () -> Unit
+    ): T = withContext(lifecycleScope.coroutineContext + Dispatchers.Main) {
+        action()
+        val childNodes = children.value.values.mapNotNull { it.nodeOrNull }
+        val childNodesOfExpectedType = childNodes.filterIsInstance<T>()
+        if (childNodesOfExpectedType.isEmpty()) {
+            throw IllegalStateException(
+                "Expected child of type [${T::class.java}] was not found after executing action. " +
+                    "Check that your action actually results in the expected child. " +
+                    "Child count: ${childNodes.size}. " +
+                    "Last child is: [${childNodes.lastOrNull()}]. " +
+                    "All children: $childNodes"
+            )
+        } else {
+            childNodesOfExpectedType.last()
+        }
+    }
+
+    protected suspend inline fun <reified T : Node> waitForChildAttached(): T =
+        withContext(lifecycleScope.coroutineContext + Dispatchers.Main) {
```

Comment:
`children` is a `Flow` so I think you can just subscribe to it and take the first result.